gallery/src/DarkModeModal/DarkModeModal.tsx

```diff
@@ -0,0 +1,111 @@
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useEffect, useState } from 'react';
+
+import {
+  NxButton,
+  NxModal,
+  NxForm,
+  NxFormGroup,
+  NxToggle,
+  useToggle,
+  NxFieldset,
+  NxRadio,
+  NxTooltip,
+  NxFontAwesomeIcon
+} from '@sonatype/react-shared-components';
+
+import { faInfoCircle } from '@fortawesome/free-solid-svg-icons';
+
+const DarkModeModal = () => {
+  const [showModal, setShowModal] = useState(false);
+  const modalCloseHandler = () => setShowModal(false);
+
+  const [enableModeChange, toggleEnableModeChange] = useToggle(false);
+  const [mode, setMode] = useState<string | null>(null);
+
+  const { classList } = document.documentElement;
+
+  useEffect(() => {
+    classList.toggle('nx-html--enable-color-schemes', enableModeChange);
+  }, [enableModeChange]);
+
+  useEffect(() => {
+    classList.toggle('nx-html--dark-mode', mode === 'dark' && enableModeChange);
+    classList.toggle('nx-html--light-mode', mode === 'light' && enableModeChange);
+  }, [enableModeChange, mode]);
+
+  return (
+    <>
+      <NxButton variant="tertiary" onClick={() => setShowModal(true)}>Theme Settings</NxButton>
+      { showModal &&
+        <NxModal id="nx-modal-dark-mode-example"
+                 onCancel={modalCloseHandler}
+                 aria-labelledby="nx-modal-dark-mode-form-header">
+          <NxForm onSubmit={modalCloseHandler}
+                  submitBtnText= "Save and Close">
+            <header className="nx-modal-header">
+              <h2 className="nx-h2" id="nx-modal-dark-mode-form-header">
+                Setting Preferences for Light or Dark Mode
+              </h2>
+            </header>
+            <div className="nx-modal-content">
+              <NxFormGroup label="Enable Theme Changes">
+                <NxToggle onChange={toggleEnableModeChange} isChecked={enableModeChange}>
+                  Opt-in to Allow Theming
+                  <NxTooltip title="Opting in will allow you to make adjustments to the display theme. Failure
+                             to do so will always result in light mode. This can be done by adding the
+                             'nx-html--enable-color-schemes' class to the HTML element.">
+                    <NxFontAwesomeIcon className="dark-mode-modal__info-icon" icon={faInfoCircle}/>
+                  </NxTooltip>
+                </NxToggle>
+              </NxFormGroup>
+              <NxFieldset label="Choose Your Mode">
+                <NxRadio name="mode"
+                         value="browserChoice"
+                         onChange={setMode}
+                         isChecked={mode === 'browserChoice'}
+                         disabled = {!enableModeChange}>
+                  Let Your Browser Color Preference Decide
```

Comment:
Wrap the text for each of these in a span, then you won't need the CSS customization.  This is why we have a general rule not to mix bare text nodes and elements as siblings unless the elements are simple inline elements.

Reply:
> Wrap the text for each of these in a span, then you won't need the CSS customization. This is why we have a general rule not to mix bare text nodes and elements as siblings unless the elements are simple inline elements.

Spans added here: [033ffd8](https://github.com/sonatype/sonatype-react-shared-components/pull/782/commits/033ffd88cf7728ebcbd52bb1eb3fddef3e7b99e7)